src/location.js

```diff
@@ -17,8 +17,29 @@ import {
   tap,
 } from 'rxjs/operators'
 
+let preventNavigationCallback = null
+let continueNavigationCallback = null
+
+export const preventNavigation = (callback) => {
+  preventNavigationCallback = callback
+  return () => {
+    preventNavigationCallback = null
+    continueNavigationCallback = null
+  }
+}
+
 export const history = createBrowserHistory({
-  getUserConfirmation: () => false,
+  getUserConfirmation: (_, historyCallback) => {
+    if (preventNavigationCallback) {
+      preventNavigationCallback(shouldContinue => {
+        if (shouldContinue) {
+          continueNavigationCallback ? continueNavigationCallback() : historyCallback(shouldContinue)
```

Comment:
Shouldn't this return the result from either one of the callback calls? I don't fully understand this change (and I don't remember all of the hairy details of this module!), but it seems to me that someone is expecting a return value from `getUserConfirmation`, and as it is, it always returns a false value (either `undefined` or `false`), no?

Reply:
Actually no, return value from this function is ignored, the expectation is for this function to call provided callback as explained in "history" repo documentation here https://github.com/remix-run/history/blob/v4/docs/Blocking.md
I guess it has "return false" just to make it more clear it had no intention of calling the callback at all? I've decided to leave it so that if your not using this new feature, code executes the same as previously.